lib/model/model.go

```diff
@@ -1319,23 +1319,25 @@ func (m *Model) closeLocked(device protocol.DeviceID) {
 	closeRawConn(conn)
 }
 
-type RequestResult struct {
+// Implements protocol.RequestResult
+type requestResult struct {
 	data    []byte
 	limiter *byteSemaphore
 }
 
-func NewRequestResult(size int, limiter *byteSemaphore) *RequestResult {
-	return &RequestResult{
+func newRequestResult(size int, limiter *byteSemaphore) *requestResult {
+	return &requestResult{
 		data:    protocol.BufferPool.Get(size),
 		limiter: limiter,
 	}
 }
 
-func (r *RequestResult) Data() []byte {
+func (r *requestResult) Data() []byte {
 	return r.data
 }
 
-func (r *RequestResult) Done() {
+// Returns the byte slice back to the pool and releases the bytes to the limiter.
+func (r *requestResult) Done() {
```

Comment:
Call close?

I am taking this from python requests library, where request returns a response and response needs to be closed to release resources.